src/main/java/org/ow2/proactive/catalog/service/BucketGrantService.java

```diff
@@ -118,8 +120,19 @@ public BucketGrantMetadata updateBucketGrantForASpecificUser(String bucketName,
             // Find the bucket grant assigned to the current user
             BucketGrantEntity bucketGrantEntity = bucketGrantRepository.findBucketGrantByUsername(bucketId, username);
             if (bucketGrantEntity != null) {
+                String oldValue = bucketGrantEntity.toString();
                 // Update the access type
                 bucketGrantEntity.setAccessType(accessType);
+                // Add modification history
+                ModificationHistoryData modificationHistoryData = new ModificationHistoryData(System.currentTimeMillis(),
+                                                                                              currentUser.getName());
+                // Set old values in history
+                modificationHistoryData.setOldValues(oldValue);
```

Comment:
I did it like this to reduce DB size. Since It's not updated yet, I found it less helpful to add the history since It already has the old values in the grant attributes. When the user edits it, I will save the old ones to show to what they have changed.

Reply:
ok I get it